agent.go

```diff
@@ -655,8 +680,7 @@ func (ep *endpoint) deleteServiceInfoFromCluster() error {
 		if n.ingress {
 			ingressPorts = ep.ingressPorts
 		}
-
-		if err := c.rmServiceBinding(ep.svcName, ep.svcID, n.ID(), ep.ID(), ep.virtualIP, ingressPorts, ep.svcAliases, ep.Iface().Address().IP); err != nil {
+		if err := c.rmServiceBinding(ep.svcName, ep.svcID, n.ID(), ep.ID(), ep.Name(), ep.virtualIP, ingressPorts, ep.svcAliases, ep.myAliases, ep.Iface().Address().IP, "deleteServiceInfoFromCluster"); err != nil {
 			return err
 		}
 	}
```

Comment:
I think you should also call deleteEndpointNameResolution from here for the case of unmanaged container and its name resolution.